<a href="https://colab.research.google.com/github/Bommireddypally/Miniprojects/blob/main/01-retail-sales-analysis/retail_sales_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Task
List the contents of the `/kaggle/input/superstore-dataset-final` directory.

## Identify CSV File

### Subtask:
List the contents of the `/kaggle/input/superstore-dataset-final` directory to find the actual CSV file within it.


**Reasoning**:
To identify the CSV file within the directory, I need to list its contents using a shell command.



In [4]:
!ls /kaggle/input/superstore-dataset-final

'Sample - Superstore.csv'


## Load Data with Correct Path

### Subtask:
Modify the `pd.read_csv` call in cell `jTUmhf84vHjy` to use the full path to the CSV file found in the previous step.


**Reasoning**:
The previous step identified 'Sample - Superstore.csv' as the correct file within the dataset directory. This step modifies the `pd.read_csv` call to use this full, correct path to load the data.



**Reasoning**:
The previous attempt to load the CSV failed with a `UnicodeDecodeError`, indicating that the file is not encoded in 'utf-8'. I will try loading it with 'latin1' encoding, which is a common encoding for such issues.



In [7]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

# Set style for better-looking plots
sns.set_style("whitegrid")
plt.rcParams['figure.figsize'] = (12, 6)

# ============================================
# STEP 1: LOAD THE DATA
# ============================================
# Download a sales dataset from Kaggle (e.g., Superstore Sales)
# Place the CSV file in the same folder as this notebook

# Load your data here
df = pd.read_csv('/kaggle/input/superstore-dataset-final/Sample - Superstore.csv', encoding='latin1')

print("Dataset loaded successfully!")
print(f"Shape: {df.shape}")
print("\nFirst few rows:")
print(df.head())

# ============================================
# STEP 2: EXPLORE THE DATA
# ============================================
print("\n" + "="*50)
print("DATA EXPLORATION")
print("="*50)

# Basic info about the dataset
print("\nDataset Info:")
print(df.info())

# Statistical summary
print("\nStatistical Summary:")
print(df.describe())

# Check for missing values
print("\nMissing Values:")
print(df.isnull().sum())

# Check column names
print("\nColumn Names:")
print(df.columns.tolist())

# ============================================
# STEP 3: DATA CLEANING
# ============================================
print("\n" + "="*50)
print("DATA CLEANING")
print("="*50)

# Remove duplicates
df = df.drop_duplicates()
print(f"Duplicates removed. New shape: {df.shape}")

# Handle missing values (example - adjust based on your data)
# df = df.dropna()  # or df.fillna(method='ffill')

# Convert date columns to datetime (adjust column name as needed)
# Example: if you have 'Order Date' column
df['Order Date'] = pd.to_datetime(df['Order Date'])
df['Ship Date'] = pd.to_datetime(df['Ship Date'])

# Extract useful date features
df['Order Year'] = df['Order Date'].dt.year
df['Order Month'] = df['Order Date'].dt.month
df['Order Month Name'] = df['Order Date'].dt.month_name()
df['Order Day'] = df['Order Date'].dt.day
df['Order Day of Week'] = df['Order Date'].dt.dayofweek # Monday=0, Sunday=6

# Verify data types after conversion
print("\nDataset Info after Date Conversion:")
print(df.info())

# ============================================
# STEP 4: EXPLORATORY DATA ANALYSIS (EDA)
# ============================================
print("\n" + "="*50)
print("EXPLORATORY DATA ANALYSIS")
print("="*50)

# Example analyses - adjust column names to match your dataset

# 1. Total Sales and Profit
print(f"\nTotal Sales: ${df['Sales'].sum():,.2f}")
print(f"Total Profit: ${df['Profit'].sum():,.2f}")
print(f"Average Order Value: ${df['Sales'].mean():,.2f}")

# 2. Sales by Category
category_sales = df.groupby('Category')['Sales'].sum().sort_values(ascending=False)
print("\nSales by Category:")
print(category_sales)

# 3. Top 10 Products by Sales
# top_products = df.groupby('Product Name')['Sales'].sum().sort_values(ascending=False).head(10)
# print("\nTop 10 Products:")
# print(top_products)

# ============================================
# STEP 5: VISUALIZATIONS
# ============================================
print("\n" + "="*50)
print("CREATING VISUALIZATIONS")
print("="*50)

# Visualization 1: Sales Trend Over Time
# monthly_sales = df.groupby('Month Name')['Sales'].sum()
# plt.figure(figsize=(12, 6))
# monthly_sales.plot(kind='line', marker='o', color='blue', linewidth=2)
# plt.title('Monthly Sales Trend', fontsize=16, fontweight='bold')
# plt.xlabel('Month', fontsize=12)
# plt.ylabel('Total Sales ($)', fontsize=12)
# plt.xticks(rotation=45)
# plt.grid(True, alpha=0.3)
# plt.tight_layout()
# plt.show()

# Visualization 2: Sales by Category (Bar Chart)
# plt.figure(figsize=(10, 6))
# category_sales.plot(kind='bar', color='skyblue', edgecolor='black')
# plt.title('Sales by Category', fontsize=16, fontweight='bold')
# plt.xlabel('Category', fontsize=12)
# plt.ylabel('Total Sales ($)', fontsize=12)
# plt.xticks(rotation=45)
# plt.tight_layout()
# plt.show()

# Visualization 3: Top 10 Products (Horizontal Bar)
# plt.figure(figsize=(10, 8))
# top_products.plot(kind='barh', color='coral', edgecolor='black')
# plt.title('Top 10 Products by Sales', fontsize=16, fontweight='bold')
# plt.xlabel('Total Sales ($)', fontsize=12)
# plt.ylabel('Product', fontsize=12)
# plt.tight_layout()
# plt.show()

# Visualization 4: Sales Distribution by Region (Pie Chart)
# region_sales = df.groupby('Region')['Sales'].sum()
# plt.figure(figsize=(8, 8))
# plt.pie(region_sales, labels=region_sales.index, autopct='%1.1f%%', startangle=90)
# plt.title('Sales Distribution by Region', fontsize=16, fontweight='bold')
# plt.tight_layout()
# plt.show()

# ============================================
# STEP 6: KEY INSIGHTS
# ============================================
print("\n" + "="*50)
print("KEY INSIGHTS")
print("="*50)

# Write your findings here after analyzing the data
insights = """
Based on the analysis, here are the key findings:

1. [Your insight about sales trends]
2. [Your insight about top-performing categories/products]
3. [Your insight about regional performance]
4. [Your insight about seasonal patterns]
5. [Your insight about profitability]

Recommendations:
- [Actionable recommendation based on your findings]
- [Another recommendation]
"""

print(insights)

# ============================================
# BONUS: ADVANCED ANALYSIS (Optional)
# ============================================

# Calculate profit margin by category
# df['Profit Margin'] = (df['Profit'] / df['Sales']) * 100
# margin_by_category = df.groupby('Category')['Profit Margin'].mean()
# print("\nProfit Margin by Category:")
# print(margin_by_category)

# Correlation analysis
# numeric_cols = df.select_dtypes(include=[np.number]).columns
# correlation = df[numeric_cols].corr()
# plt.figure(figsize=(10, 8))
# sns.heatmap(correlation, annot=True, cmap='coolwarm', center=0)
# plt.title('Correlation Heatmap', fontsize=16, fontweight='bold')
# plt.tight_layout()
# plt.show()

print("\n" + "="*50)
print("Analysis Complete! ✨")
print("="*50)
print("\nNext steps:")
print("1. Uncomment and adjust the code sections based on your dataset columns")
print("2. Run each section step by step")
print("3. Document your findings in the insights section")
print("4. Save your visualizations")
print("5. Create a README.md for your GitHub repository")

Dataset loaded successfully!
Shape: (9994, 21)

First few rows:
   Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
1       2  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
2       3  CA-2016-138688   6/12/2016   6/16/2016    Second Class    DV-13045   
3       4  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
4       5  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       

## Final Task

### Subtask:
Confirm that the dataset is loaded successfully and display the first few rows to verify.


## Summary:

### Data Analysis Key Findings

*   The correct CSV file, `Sample - Superstore.csv`, was successfully identified within the `/kaggle/input/superstore-dataset-final` directory.
*   The dataset was successfully loaded using `pd.read_csv` by specifying `encoding='latin1'`, which resolved an initial `UnicodeDecodeError`.
*   The loaded dataset `df` contains 9,994 rows and 21 columns.
*   Initial data quality checks revealed no missing values and no duplicate rows after an attempt to remove them, confirming the dataset is relatively clean for further analysis.

### Insights or Next Steps

*   Always verify the correct file path and consider different encoding options (like 'latin1') when encountering `UnicodeDecodeError` during CSV loading.
*   With the data successfully loaded and basic cleaning confirmed, the next logical steps involve performing detailed Exploratory Data Analysis (EDA) and creating visualizations to uncover trends and patterns in the Superstore sales data.
